src/count.cpp

```diff
@@ -0,0 +1,129 @@
+#include <iostream>
+#include <vector>
+#include <string>
+#include <cstring>
+#include "parser.h"
+#include "utils.h"
+#include <memory>
+extern "C"
+{
+#include <tree_sitter/api.h>
+}
+
+// update a boolean if only the cur node is a function call node
+void check_call_node(const char *type, bool &is_parent_call) {
+    // types of function nodes: call, call_expression, binary_expression(eg: std::cout << "hello")
+    if ((!strcmp("call", type) || !strcmp("call_expression", type) || !strcmp("binary_expression", type)))
+    {
+        is_parent_call = true;
+    }
+}
+
+// check if the cur node is a function definition node
+void check_func_def_node(const char *type, bool &is_parent_func_def) {
+    if ((!strcmp("function_definition", type)))
+    {
+        is_parent_func_def = true;
+    }
+}
+
+// name of the function called is met in the node which is met before meeting the arguement list node
+void check_call_identifier_end(const std::string identifier, const std::string &feature, const char *type, int &count, bool &is_parent_call) {
+    if ((!strcmp("argument_list", type) || !strcmp("<<", type)) && is_parent_call)
```

Comment:
What is the `"<<"` check for here?

This isn't necessarily a requested change--just a question.

Reply:
It is used to catch `std::cout << `